packages/renderer/src/lib/table/Table.svelte

```diff
@@ -0,0 +1,165 @@
+<style>
+.grid-table {
+  display: grid;
+}
+</style>
+
+<script lang="ts">
+/* eslint-disable import/no-duplicates */
+// https://github.com/import-js/eslint-plugin-import/issues/1479
+import { afterUpdate, tick } from 'svelte';
+import Checkbox from '../ui/Checkbox.svelte';
+import type { Column, Row } from './table';
+import { flip } from 'svelte/animate';
+/* eslint-enable import/no-duplicates */
+
+export let kind: string;
+export let data: any[];
+export let columns: Column<any>[];
+export let row: Row<any>;
+
+// number of selected items in the list
+export let selectedItemsNumber: number = 0;
+$: selectedItemsNumber = row.selectable
+  ? data.filter(object => row.selectable?.(object)).filter(object => object.selected).length
+  : 0;
+
+// do we need to unselect all checkboxes if we don't have all items being selected ?
+$: selectedAllCheckboxes = row.selectable
+  ? data.filter(object => row.selectable?.(object)).every(object => object.selected)
+  : false;
+
+function toggleAll(checked: boolean) {
+  if (!row.selectable) {
+    return;
+  }
+  const toggleData = data;
+  toggleData.filter(object => row.selectable?.(object)).forEach(object => (object.selected = checked));
+  data = toggleData;
+}
+
+let sortCol: Column<any>;
+let sortAscending: boolean;
+
+function sort(column: Column<any>) {
+  if (!column) {
+    return;
+  }
+
+  let comparator = column.comparator;
+  if (!comparator) {
+    // column is not sortable
+    return;
+  }
+
+  if (sortCol === column) {
+    sortAscending = !sortAscending;
+  } else {
+    sortCol = column;
+    sortAscending = true;
+  }
+  if (!sortAscending) {
+    // we're already sorted, switch to reverse order
+    let comparatorTemp = comparator;
+    comparator = (a, b) => -comparatorTemp(a, b);
+  }
+  const sortedData = data;
+  sortedData.sort(comparator);
+  data = sortedData;
+}
+
+afterUpdate(async () => {
+  await tick();
+  setGridColumns();
+});
+
+function setGridColumns() {
+  // section and checkbox columns
+  let columnWidths: string[] = ['20px', '32px'];
+
+  // custom columns
+  for (const column of columns) {
+    if (column.info.width) {
+      columnWidths.push(column.info.width);
+    } else {
+      columnWidths.push('1fr');
+    }
+  }
```

Comment:
You could use a map here:

(not tested)
```
columnWidths = columns.map(c => c.info.width ?? '1fr');
```

Reply:
Yup, updated in latest commit.